pkgs/development/misc/juce/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+
+# Native build inputs
+, cmake
+, pkg-config
+, makeWrapper
+
+# Dependencies
+, alsa-lib
+, freetype
+, curl
+, libglvnd
+, webkitgtk
+, pcre
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "juce";
+  version = "7.0.7";
+
+  src = fetchFromGitHub {
+    owner = "juce-framework";
+    repo = "juce";
+    rev = finalAttrs.version;
+    hash = "sha256-r+Wf/skPDexm3rsrVBoWrygKvV9HGlCQd7r0iHr9avM=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    makeWrapper
+  ];
+
+  buildInputs = [
+    freetype # libfreetype.so
+    curl # libcurl.so
+    alsa-lib # libasound.so
+    libglvnd # libGL.so
+    stdenv.cc.cc.lib # libstdc++.so libgcc_s.so
+    webkitgtk # webkit2gtk-4.0
+    pcre # libpcre2.pc
+  ];
+
+  postInstall = ''
+    mv $out/bin/JUCE-${finalAttrs.version}/* $out/bin
+    rmdir $out/bin/JUCE-${finalAttrs.version}
+    rm -rf $out/include
+    rm -rf $out/lib
```

Comment:
Why?

Reply:
https://github.com/NixOS/nixpkgs/pull/253561#issuecomment-1709883872
https://github.com/NixOS/nixpkgs/pull/253561#pullrequestreview-1613376032

I'm indifferent on it if it's useful. 

I had thought that the $out/lib directory may be where some exposed library headers were as the [original issue pointed out](https://github.com/NixOS/nixpkgs/issues/252737#issuecomment-1709671359) however i never got docs back when i asked about it

Reply:
I can hardly imagine that packages with juce as a dependency don't use these libraries and header files. It would be interesting to compare the list of files with [that of Arch Linux](https://archlinux.org/packages/extra/x86_64/juce/files/).

Reply:
I agree, I'll remove the cleanup of `$out/lib`

Reply:
fixed in https://github.com/NixOS/nixpkgs/pull/253561/commits/1b4aad26453a5812339c04bb3a3019618603814a